### PR TITLE
statistics: support dropping partition/global level statistics

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary: statistics: support dropping partition/global level statistics

### What is changed and how it works?

Must merge https://github.com/pingcap/parser/pull/1177 first.
Support dropping partition/global level statistics.

### Check List 

Tests 

- Unit test

### Release note 

- statistics: support dropping partition/global level statistics

### CODE REVIEW

**Review by rebelice — APPROVED** [review]

**Review by Reminiscent — APPROVED** [review]
/LGTM

File: executor/simple_test.go [hunk 1]
```diff
@@ -599,6 +600,45 @@ func (s *testFlushSuite) TestFlushPrivilegesPanic(c *C) {
...
+	tk.MustExec("use test")
+	tk.MustExec(`create table t (
+	a int,
+	key(a)
+)
+partition by range (a) (
+	partition p0 values less than (10),
+	partition p1 values less than (20)
+)`)
+	tk.MustExec("set @@tidb_partition_prune_mode='dynamic-only'")
```
Comment (Reminiscent) [inline executor/simple_test.go:614]: Should be `dynamic` here.

**Review by Reminiscent — APPROVED** [review]
/LGTM

**Pull request merged**